### PR TITLE
Update datePickerAndroid bindings

### PR DESCRIPTION
**Pull request by rodan-lewarx**
datePickerAndroid bindings now use bs.deriving abstract and bs.deriving jsConverter

### CODE REVIEW

File: src/datePickerAndroid.re [hunk 1]
```diff
@@ -19,48 +17,45 @@ type action =
...
-    "minDate": Js.Nullable.fromOption(minDate),
-    "maxDate": Js.Nullable.fromOption(maxDate),
-    "mode":
-      switch (mode) {
-      | Default => "default"
-      | Calendar => "calendar"
-      | Spinner => "spinner"
-      },
-  })
+let open_ = (~date: Js.Date.t, ~minDate=?, ~maxDate=?, ~mode=`default, ()) =>
```
Comment (wokalski) [inline src/datePickerAndroid.re]: cool!

File: src/datePickerAndroid.rei [hunk 2]
```diff
@@ -8,10 +8,7 @@ type action =
   | Dismissed
   | Set(response);
 
-type mode =
-  | Calendar
-  | Spinner
-  | Default;
+type mode = [ | `calendar | `spinner | `default];
```
Comment (wokalski) [inline src/datePickerAndroid.rei]: I'm not sure about this change. I don't think it's justified to change the public API in this case.

Comment (rodan-lewarx) [reply src/datePickerAndroid.rei]: @wokalski Sorry I didn't think about that. I am just using abstract and jsConverter everywhere. 

I just wanted to do the same I did for timePickerAndroid https://github.com/reasonml-community/bs-react-native/pull/233 If we are not using jsConverter here, should I change timePickerAndroid to use the same pattern as used here?

Comment (wokalski) [reply src/datePickerAndroid.rei]: Hmm, ok, that's a good point. Let's make this breaking change then!

**Comment by wokalski** [comment]
Thanks a lot @rodan-lewarx 👍

**Pull request merged**